angel_system/global_step_prediction/global_step_predictor.py

```diff
@@ -242,69 +344,71 @@ def process_new_confidences(self, activity_confs):
                 activated_indexes,
-                np.where(
+                np.nonzero(
                     self.activated_activities[:, 1]
                     >= self.deactivate_thresh_frame_count
                 )[0],
             )
             if 1 in flipping_off_indexes:
                 print(f"boo i = {i}")
-            # flipping_off_indexes = np.where((self.activated_activities[:,0] == 1,)
-            #    and (self.activated_activities[:,1] <= self.
-            #        deactivate_thresh_frame_count,))[1]
             self.activated_activities[flipping_off_indexes, 0] = 0
             self.activated_activities[flipping_off_indexes, 1] = 0
 
             # Now, go through ACTIVE trackers and see which corresponds to "flipping_on_indexes."
             for tracker_ind, tracker in enumerate(self.trackers):
-                current_step = tracker["current_step"]
+                current_granular_step = tracker["current_granular_step"]
 
                 # TODO: For now the tracker can jump 1 or 2 steps, if base
                 # jump criteria is met. Add "weak" threshold too.
-                if current_step == tracker["total_num_steps"] - 1:
+                if current_granular_step == tracker["total_num_granular_steps"] - 1:
                     continue
 
-                next_step = current_step + 1
-                next_next_step = min(next_step + 1, tracker["total_num_steps"] - 1)
-                current_activity = tracker["step_to_activity_id"][current_step]
-                next_activity = tracker["step_to_activity_id"][next_step]
-                try:
-                    next_next_activity = tracker["step_to_activity_id"][next_next_step]
-                except:
-                    import ipdb
-
-                    ipdb.set_trace()
+                next_granular_step = current_granular_step + 1
+                next_next_granular_step = min(
+                    next_granular_step + 1, tracker["total_num_granular_steps"] - 1
+                )
+                current_activity = tracker["granular_step_to_activity_id"][
+                    current_granular_step
+                ]
+                next_activity = tracker["granular_step_to_activity_id"][
+                    next_granular_step
+                ]
+                next_next_activity = tracker["granular_step_to_activity_id"][
+                    next_next_granular_step
+                ]
 
                 # TODO: prioritize a 1-step jump over a 2-step jump. Create a
                 # second loop just for the 2-step jumps, after this loop has completed
                 # searching for one-step jumps.
                 if next_activity in flipping_on_indexes:
-                    self.trackers[tracker_ind]["current_step"] = next_step
+                    try:
+                        self.increment_granular_step(tracker_ind)
+                    except:
+                        import ipdb
+
+                        ipdb.set_trace()
```

Comment:
`ipdb` is not currently a package dependency so this will error if an exception occurs. Did you possibly mean to remove this "breakpoint"?

Reply:
Definitely. Oops.